src/SMSAero/API.hs

```diff
@@ -329,32 +278,35 @@ instance ToJSON SendResponse where
   toJSON SendNoCredits = object
     [ "result" .= ("no credits" :: Text)]
 
--- | Helper to define @fromText@ matching @toText@.
-boundedFromText :: (Enum a, Bounded a, ToText a) => Text -> Maybe a
-boundedFromText = flip lookup xs
+-- | Helper to define @parseUrlPiece@ matching @toUrlPiece@.
+boundedParseUrlPiece :: (Enum a, Bounded a, ToHttpApiData a) => Text -> Either Text a
+boundedParseUrlPiece x = lookupEither ("could not parse: " <> x) x xs
```

Comment:
Please open a PR against https://github.com/fizruk/http-api-data to add this and similar functions there.